### DISCUSSION TITLE
Microsoft.Data.SQLClient Failed to connect to SQL Server

### DISCUSSION THREAD

**Discussion by chandra-arifin** [dotnet/SqlClient#1690] [Q&A]
i know this is not recommended to "Directly" connect to SQL Server (Localhost) with Android (MAUI in this context).
but is it a bug or can add this features?

my context is:
1. windows 10 (VS 2022 Preview 6)
2. Android (use MAUI)
3. SQL Server 2019 (Localhost) and connect to LAN via WIFI

i can connect to sql server with windows only (WinUI) , but refused to work with android.
my question is:
is it possible to use microsoft.data.sqlclient to connect directly from android?

info:
in 2019, i use android studio and build apps to directly connect to sql server via WIFI, and works fine.
(no TLS or encryption on database)

with this jTds library:
http://jtds.sourceforge.net/

this library works, but i don't know if it is can work with MAUI. not trial this yet.
but this library refused to work with Sql Azure (With TLS >= 1.0 on)

my suspects is the library still work with SSL (No Encryption), and maybe Microsoft.data.sqlclient can do this in the future?

thanks

**Comment by JRahnama** [comment]
@chandra-arifin sorry for the delay. There are several issues with MAUI trying to get connection from SQL server, #1662. #1656, probably #1412. As you have mentioned it is not safe to use localhost directly on the device and it is better to have one of the services to be used in between such as ASP.NET, but just for the curiosity sake what version of the M.D.SqlClient you used? Is it possible to test with the latest preview version?

**Comment by chandra-arifin** [comment]
Latest stable mdsqlclient 4.1, and preview 5. Too. But no luck. Because in my city, only few users. So dont want to use other service layer (web api), i remember i can connect with ADO.Net directly via Wifi connection maybe at 2009 when ipad 1 launch and samsung at that time. I try it with samsung gadget. And connect to localhost for testing successfully. And then abandon the project, because no one use it.

And curious about MAUI, then try it agan when GA. And now use MAUI.Markup extention happily. Thanks for ur respond.


Get Outlook for Android<https://aka.ms/ghei36>
________________________________
